src/geocel/rasterize/ImageData.hh

```diff
@@ -7,28 +7,128 @@
 //---------------------------------------------------------------------------//
 #pragma once
 
+#include "corecel/Assert.hh"
 #include "corecel/cont/Array.hh"
-#include "corecel/cont/Span.hh"
+#include "corecel/data/Collection.hh"
+#include "corecel/data/CollectionBuilder.hh"
 #include "geocel/Types.hh"
 
 namespace celeritas
 {
 //---------------------------------------------------------------------------//
 /*!
- * Construction arguments for an on-device image view.
+ * Scalar properties for building a rasterized image.
+ *
+ * These properties specify a "window" that's a slice of a 3D geometry. It uses
+ * graphics conventions of making the upper left corner the origin.
+ *
+ * The \c down basis vector corresponds to increasing \em j and is used for
+ * track initialization.  The \c right basis vector corresponds to increasing
+ * \em i and is used for track movement. Because the user-specified window may
+ * not have an integer ratio of the two sides, we have a "max length" for
+ * raytracing to the right. This also lets us round up the image dimensions to
+ * a convenient alignment.
+ *
+ * All units are "native" length.
  */
-struct ImageData
+struct ImageParamsScalars
 {
-    Real3 origin;  //!< Upper left corner
-    Real3 down_ax;  //!< Downward axis (increasing j, track initialization)
-    Real3 right_ax;  //!< Rightward axis (increasing i, track movement)
-    real_type pixel_width;  //!< Width of a pixel
-    Array<unsigned int, 2> dims;  //!< Image dimensions (j, i)
-    Span<int> image;  //!< Stored image [j][i]
+    using Size2 = Array<size_type, 2>;
+
+    Real3 origin{};  //!< Upper left corner
+    Real3 down{};  //!< Downward basis vector
+    Real3 right{};  //!< Rightward basis vector (increasing i, track movement)
+    real_type pixel_width{};  //!< Width of a pixel
+    Size2 dims{};  //!< Image dimensions (j, i)
+    real_type max_length{};  //!< Maximum distance along rightward to trace
```

Comment:
Your Scalars structure contain several vectors!

Reply:
Ha good point, I thought about that... I named it like we do for the other "runtime constants" elsewhere in the code. I've added a note on https://github.com/celeritas-project/celeritas/issues/721 to rename `Scalars`→`Constants` everywhere.